toml/src/main/kotlin/org/rust/toml/inspections/CargoTomlCyclicFeatureInspection.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.inspections
+
+import com.intellij.codeInspection.InspectionManager
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.findDescendantOfType
+import org.rust.cargo.CargoConstants
+import org.toml.lang.psi.*
+
+/**
+ *  * Consider `Cargo.toml`:
+ * ```
+ * [features]
+ * foo = ["foo"]
+ *       #^ Shows error that "foo" feature depends on itself
+ * ```
+ */
+class CargoTomlCyclicFeatureInspection : LocalInspectionTool() {
+    override fun checkFile(file: PsiFile, manager: InspectionManager, isOnTheFly: Boolean): Array<ProblemDescriptor?>? {
```

Comment:
With the local crate index, there will probably be multiple new inspections for `Cargo.toml` files. I wonder if it would be better to create a Toml PSI visitor so we could implement inspections using the `buildVisitor` method? Assuming it has benefits versus `checkFile` (such as better performance), otherwise it probably doesn't matter.

Reply:
Yep, this definitely affects inspection latency

Reply:
I tried to create a visitor in https://github.com/intellij-rust/intellij-rust/pull/6578.